ols/src/cache/postgres_cache.py

```
@@ -388,21 +388,31 @@ def set_topic_summary(
         Returns:
             True if the cache is ready, False otherwise.
         """
-        # TODO: when the connection is closed and the database is back online,
-        # we need to reestablish the connection => implement this
         if not self.connection or self.connection.closed == 1:
-            return False
+            try:
+                logger.info("Detected dead connection, attempting reconnect")
+                self.connect()
+                return True
+            except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)